File: Ejercicio 14/Ejercicio 14.cpp
#include <iostream>
#include <cstdlib>
#include <ctime>

using namespace std;
//  Listas Enlazadas
struct Nodo {
    int valor;
    Nodo* siguiente;
};

int main() {
    srand(time(0));
    
    Nodo* cabeza = nullptr;
    Nodo* actual = nullptr;
    
    cout << "Generando 15 valores aleatorios..." << endl;
    
    // Crear 15 nodos con valores aleatorios
    for (int i = 0; i < 15; i++) {
        Nodo* nuevoNodo = new Nodo;
        nuevoNodo->valor = 1 + rand() % 100;
        nuevoNodo->siguiente = nullptr;
        
        if (cabeza == nullptr) {
            cabeza = nuevoNodo;
            actual = cabeza;
        } else {
            actual->siguiente = nuevoNodo;
            actual = nuevoNodo;
        }
    }
    
    // Mostrar la lista
    cout << "Lista Enlazada:" << endl;
    actual = cabeza;
    int contador = 1;
    while (actual != nullptr) {
        cout << "Nodo " << contador << ": " << actual->valor << endl;
        actual = actual->siguiente;
        contador++;
    }
    
    // Liberar memoria
    actual = cabeza;
    while (actual != nullptr) {
        Nodo* temp = actual;
        actual = actual->siguiente;
        delete temp;
    }
    
    return 0;
}

File: Ejercicio 14/Ejercicio 14.cs
using System;

class Program
{
    // listas enlazadas
    class Nodo
    {
        public int valor;
        public Nodo siguiente;
    }

    static void Main()
    {
        Random rand = new Random();
        
        Nodo cabeza = null;
        Nodo actual = null;
        
        Console.WriteLine("Generando 15 valores aleatorios");
        Console.WriteLine(" ");
        // Crear 15 nodos con valores aleatorios
        for (int i = 0; i < 15; i++)
        {
            Nodo nuevoNodo = new Nodo();
            nuevoNodo.valor = 1 + rand.Next(100); // 1-100
            nuevoNodo.siguiente = null;
            
            if (cabeza == null)
            {
                cabeza = nuevoNodo;
                actual = cabeza;
            }
            else
            {
                actual.siguiente = nuevoNodo;
                actual = nuevoNodo;
            }
        }
        
        // Mostrar la lista
        Console.WriteLine("Lista Enlazada:");
        actual = cabeza;
        int contador = 1;
        while (actual != null)
        {
            Console.WriteLine($"Nodo {contador}: {actual.valor}");
            actual = actual.siguiente;
            contador++;
        }
    }
}